x-pack/plugins/transform/public/app/sections/create_transform/components/step_details/step_details_form.tsx

```diff
@@ -397,51 +402,82 @@ export const StepDetailsForm: FC<Props> = React.memo(
+                  defaultMessage: 'Destination index',
+                })}
+                isInvalid={!indexNameEmpty && !indexNameValid}
+                helpText={
+                  indexNameExists &&
+                  i18n.translate('xpack.transform.stepDetailsForm.destinationIndexHelpText', {
+                    defaultMessage:
+                      'An index with this name already exists. Be aware that running this transform will modify this destination index.',
+                  })
                 }
-              )}
-              isInvalid={!indexNameEmpty && !indexNameValid}
-              data-test-subj="transformDestinationIndexInput"
-            />
-          </EuiFormRow>
+                error={
+                  !indexNameEmpty &&
+                  !indexNameValid && [
+                    <Fragment>
+                      {i18n.translate(
+                        'xpack.transform.stepDetailsForm.destinationIndexInvalidError',
+                        {
+                          defaultMessage: 'Invalid destination index name.',
+                        }
+                      )}
+                      <br />
+                      <EuiLink href={esIndicesCreateIndex} target="_blank">
+                        {i18n.translate(
+                          'xpack.transform.stepDetailsForm.destinationIndexInvalidErrorLink',
+                          {
+                            defaultMessage: 'Learn more about index name limitations.',
+                          }
+                        )}
+                      </EuiLink>
+                    </Fragment>,
+                  ]
+                }
+              >
+                <EuiFieldText
+                  value={destinationIndex}
+                  onChange={(e) => setDestinationIndex(e.target.value)}
+                  aria-label={i18n.translate(
+                    'xpack.transform.stepDetailsForm.destinationIndexInputAriaLabel',
+                    {
+                      defaultMessage: 'Choose a unique destination index name.',
+                    }
+                  )}
+                  isInvalid={!indexNameEmpty && !indexNameValid}
+                  data-test-subj="transformDestinationIndexInput"
+                />
+              </EuiFormRow>
+            </EuiFlexItem>
+            {stepDefineState.transformFunction === TRANSFORM_FUNCTION.LATEST ? (
+              <EuiFlexItem>
+                <EuiCallOut color="warning" iconType="alert" size="m">
+                  <p>
+                    <FormattedMessage
+                      id="xpack.transform.stepDetailsForm.destinationIndexWarning"
+                      defaultMessage="Note that dynamic mappings based on the source fields will be used if the destination index does not exist. If alternate mappings are required, use the {docsLink} prior to starting the transform. If the transform fails, please check the messages tab for the transform on the Stack Management page for errors."
```

Comment:
I altered the first sentence a bit and shortened the last sentence, but if it's altered the technical accuracy, let me know!
```suggestion
                      defaultMessage="If the destination index does not exist, the transform uses dynamic mappings based on the source fields. If alternate mappings are required, use the {docsLink} prior to starting the transform. If the transform fails, check the messages tab on the Stack Management page for errors."
```

Reply:
Updated in https://github.com/elastic/kibana/pull/87858/commits/8bb38f53b71c5605d2668516fad7d0dd4ac9ff77